### DISCUSSION TITLE
Is there any way to retrieve bookmarks?

### DISCUSSION THREAD

**Discussion by rbielaws** [audiamus/BookLibConnect#123] [Q&A]
I have several hundred books and listened to at least a few hundred of them with the audible app.  While listening I often made notes using the bookmark feature.  I've long wanted to collect these notes but can't seem to find a practical way.  I don't see them in any of the JSON files but thought it might be worth asking if a method exists.
I do know I can begin listening to every book one by one from a browser and use the play window's menu to display them so I can copy and paste them somewhere, as you can imagine, this would be an immensely tedious process.

**Comment by audiamus** [comment]
According to @mkb79, such a method exists: [Audible 0.9.1 documentation](https://audible.readthedocs.io/en/latest/misc/external_api.html), but I have not looked at it myself. I never used any of the official Audible apps to listen to a book and hence did not create any bookmarks.

**Reply by rbielaws** [reply]
Thanks.

GET https://cde-ta-g7g.amazon.com/FionaCDEServiceEngine/sidecar Returns the clips, notes and bookmarks of a book.
Query Parameters:
type (string) – [“AUDI”]
key (string) – asin of the book

If I get something working I'll post back how I did it, in case someone else finds this thread!
